Group1

abdulrhman500 opened the pull request.

abdulrhman500 committed:
chore: setup & eventhandler

abdulrhman500 committed:
chore: setup & eventhandler

abdulrhman500 committed:
chore: models

abdulrhman500 committed:
chore(setup):docker

abdulrhman500 committed:
Merge branch 'dev' of https://github.com/Podzilla/erp into dev

abdulrhman500 committed:
feat: revenue summary

abdulrhman500 committed:
feat: revenue/by-category & products/top-sellers

abdulrhman500 committed:
fix: naming convention fix and validation util

abdulrhman500 committed:
chore: pom.xml

abdulrhman500 committed:
chore: merge with dev to keep up

abdulrhman500 committed:
chore: merge with dev

abdulrhman500 committed:
test: getTopSellers, getRevenueSummary, getRevenueByCategory

abdulrhman500 committed:
tests(RevenueReportServiceIntegrationTest): integration test

abdulrhman500 committed:
tests(RevenueReportServiceIntegrationTest): integration test

abdulrhman500 committed:
tests(ProductAnalyticsServiceIntegrationTest): still 3 fails

abdulrhman500 committed:
tests(ProductAnalyticsServiceIntegrationTest): all pass

Mohamed-Khaled308 commented on src/main/java/com/Podzilla/analytics/api/DTOs/RevenueSummaryRequest.java:
```diff
@@ -0,0 +1,25 @@
+@AllArgsConstructor
+@Builder
+public class RevenueSummaryRequest {
+    private LocalDate startDate;
```
please add the validation logic and swagger docs for request dtos. 
for example use @NotNull, @DateTimeFormat and @Schema for docs

Mohamed-Khaled308 commented on src/main/java/com/Podzilla/analytics/api/controllers/ProductReportController.java:
```diff
@@ -1,14 +1,60 @@
+
+    @GetMapping("/top-sellers") 
+    public ResponseEntity<List<TopSellerResponse>> getTopSellers(
+        @RequestParam @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) LocalDate startDate,
```
Use the request DTO as a whole instead of passing each field one by one. Spring will fill it in for you. Also, put the validation rules inside the DTO, not here.

Mohamed-Khaled308 commented on src/main/java/com/Podzilla/analytics/utils/ValidationUtils.java:
```diff
@@ -0,0 +1,49 @@
+import java.time.LocalDate;
+import org.springframework.http.ResponseEntity;
+
+public class ValidationUtils {
```
We don’t need this class anymore. You can use Jakarta Validation annotations like @NotNull or @Min directly in the request DTO. If you need a custom check, create a custom annotation for it. In any case, all validation should go inside the request DTO.

abdulrhman500 committed:
fix: dtos validation and docs

abdulrhman500 committed:
fix: remove shema.sql

abdulrhman500 committed:
fix: sql query error in findRevenueSummary

abdulrhman500 committed:
fix: use dto in ProductReportController

abdulrhman500 committed:
style: clean up code formatting and remove redundant comments

- Remove trailing whitespace and ensure consistent newline endings
- Remove redundant comments in RevenueSummaryRequest and ErrorResponse

abdulrhman500 committed:
Merge branch 'dev' into group1

abdulrhman500 committed:
Update pom.xml

abdulrhman500 committed:
Update pom.xml

abdulrhman500 committed:
refactor: clean up code formatting and remove unused files

Removed unused repositories, models, and utility files. Cleaned up code formatting in multiple files, including fixing indentation, removing redundant imports, and ensuring consistent spacing. Commented out test file for future reference. These changes improve code maintainability and readability without altering functionality.

abdulrhman500 committed:
style: fix char count

Adjust indentation in RevenueReportService and OrderRepository to maintain consistent code style

abdulrhman500 committed:
style(repository): standardize SQL keyword case in OrderRepository

Ensure consistent use of uppercase for SQL keywords (e.g., SELECT, FROM, WHERE) to improve code readability and maintainability.

abdulrhman500 committed:
refactor(repositories): convert multi-line SQL queries to single-line format

abdulrhman500 commented:
@Mohamed-Khaled308

Mohamed-Khaled308 committed:
chore: apply @ValidDateRange and pass attributes directly to service

Mohamed-Khaled308 committed:
chore: clean up and deduplicate Maven dependencies
